fang/src/asynk/async_queue.rs

```diff
@@ -156,375 +136,289 @@ pub trait AsyncQueueable: Send {
+            _ => BackendSqlX::NoBackend,
+        };
+
+        self.backend = backend;
+
+        self.pool = Some(pool);
+        self.connected = true;
+        Ok(())
     }
 
     async fn fetch_and_touch_task_query(
-        transaction: &mut Transaction<'_>,
+        pool: &Pool<Any>,
+        backend: &BackendSqlX,
         task_type: Option<String>,
     ) -> Result<Option<Task>, AsyncQueueError> {
         let task_type = match task_type {
             Some(passed_task_type) => passed_task_type,
             None => DEFAULT_TASK_TYPE.to_string(),
         };
 
-        let task = match Self::get_task_type_query(transaction, &task_type).await {
-            Ok(some_task) => Some(some_task),
-            Err(_) => None,
-        };
+        let query_params = QueryParams::builder().task_type(&task_type).build();
+
+        let task = backend
+            .execute_query(SqlXQuery::FetchTaskType, pool, query_params)
+            .await
+            .map(|val| val.unwrap_task())
+            .ok();
+
         let result_task = if let Some(some_task) = task {
-            Some(
-                Self::update_task_state_query(transaction, &some_task, FangTaskState::InProgress)
-                    .await?,
-            )
+            let query_params = QueryParams::builder()
+                .uuid(&some_task.id)
+                .state(FangTaskState::InProgress)
+                .build();
+
+            let task = backend
+                .execute_query(SqlXQuery::UpdateTaskState, pool, query_params)
+                .await?
+                .unwrap_task();
+
+            Some(task)
         } else {
             None
         };
         Ok(result_task)
     }
 
-    async fn get_task_type_query(
-        transaction: &mut Transaction<'_>,
-        task_type: &str,
-    ) -> Result<Task, AsyncQueueError> {
-        let row: Row = transaction
-            .query_one(FETCH_TASK_TYPE_QUERY, &[&task_type, &Utc::now()])
-            .await?;
-
-        let task = Self::row_to_task(row);
-
-        Ok(task)
-    }
-
-    async fn update_task_state_query(
-        transaction: &mut Transaction<'_>,
-        task: &Task,
-        state: FangTaskState,
-    ) -> Result<Task, AsyncQueueError> {
-        let updated_at = Utc::now();
-
-        let row: Row = transaction
-            .query_one(UPDATE_TASK_STATE_QUERY, &[&state, &updated_at, &task.id])
-            .await?;
-        let task = Self::row_to_task(row);
-        Ok(task)
-    }
-
     async fn insert_task_query(
-        transaction: &mut Transaction<'_>,
-        metadata: serde_json::Value,
-        task_type: &str,
-        scheduled_at: DateTime<Utc>,
-    ) -> Result<Task, AsyncQueueError> {
-        let row: Row = transaction
-            .query_one(INSERT_TASK_QUERY, &[&metadata, &task_type, &scheduled_at])
-            .await?;
-        let task = Self::row_to_task(row);
-        Ok(task)
-    }
-
-    async fn insert_task_uniq_query(
```

Comment:
what about this function? why was it removed?

Reply:
I think it was replaced by `insert_task if_not_exists`